Sync

dbolcioni opened the pull request.

mbrukman committed:
Fix spelling.

Signed-off-by: Misha Brukman <mbrukman@google.com>

mbrukman committed:
Remove a reference to an invalid repo.

For one, this repo is actually https://github.com/StartTheShift/titan-tools but
was automatically updated to https://github.com/StartTheShift/janusgraph-tools
during the Great Renaming process, so it's already incorrect.

That said, it was last updated in 2013 and claims to only work with Titan 0.2.1,
so I think it's safe to assume it may not work with either recent versions of
Titan or any versions of JanusGraph, so it may be best to avoid pointing users
at it to avoid data corruption.

Signed-off-by: Misha Brukman <mbrukman@google.com>

mbrukman committed:
Switch to recommended Asciidoc header style.

As described in the Asciidoc best practices document:
http://asciidoctor.org/docs/asciidoc-recommended-practices/#section-titles
using heading markers using underlines of various characters is not recommended
for several reasons:

* hard to remember which character means which level header
* hard to maintain the correct number of characters (must match heading line)

We learned this the hard way during the global Titan -> JanusGraph rewrite that
this breaks headings, because "JanusGraph" is longer than "Titan".

This global change was implemented via this Python script:

```python
#!/usr/bin/python

import re
import sys

def main(argv):
    # http://asciidoctor.org/docs/asciidoc-recommended-practices/#section-titles
    patterns = [
        (re.compile('^=+$'), '='),
        (re.compile('^-+$'), '=='),
        (re.compile('^~+$'), '==='),
        (re.compile('^\^+$'), '===='),
        (re.compile('^\++$'), '====='),
    ]

    with open(argv[1], 'r') as input_file:
        prev_line = None
        curr_line = None
        for line in input_file.readlines():
            prev_line = curr_line
            curr_line = line

            if prev_line is None:
                continue

            for pattern, heading in patterns:
                if pattern.match(curr_line) and len(prev_line) == len(curr_line):
                    sys.stdout.write('%s %s' % (heading, prev_line))
                    prev_line = None
                    curr_line = None
                    break

            if prev_line is not None:
                sys.stdout.write(prev_line)

        # end for
        if curr_line is not None:
            sys.stdout.write(curr_line)

    # end with

if __name__ == '__main__':
    main(sys.argv)
```

and this simple shell script driver:

```bash
#!/bin/bash

tmp="$(mktemp /tmp/asciidoc-cleanup.XXXXXX)"
for file in *.txt ; do
  ./asciidoc.py $file > $tmp
  if [[ $? == 0 ]]; then
    mv $tmp $file
  else
    echo "error in $file" >&2
  fi
done
```

Signed-off-by: Misha Brukman <mbrukman@google.com>

lindsayismith committed:
Modifications to HBase storage to support Google Cloud Bigtable.

1. Moved getRegionLocations to the ConnectionMask interface. Implemented
in 0_98 implementation using the existing method, in 1_0 implementation
uses connection.getRegionLocator method instead.

2. Cache the returned value of HBaseStorageManager.getDeployement so
that subsequent calls are fast.  This is called a lot during queries,
and this requires a backend call, so is expensive. Using Bigtable
without this cache you quickly hit the quota limits for ListTables.

Tested by running full test suite in the janusgraph-hbase-parent folder.

Change ConnectionMask interface to use 1.0 style HRegionLocations as the return type.  Update normalizeKeyBounds to operate on this list.

Add documentation for connecting to Bigtable backend

rafatuita committed:
Implemented backward compatibility with titan

pluradj-ibm committed:
Add repositories to enable dependency reports

Signed-off-by: Jason Plurad <pluradj@us.ibm.com>

pluradj committed:
Merge pull request #137 from pluradj/project-info-reports

Add repositories to enable dependency reports

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/janusgraph?pullRequest=1) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber/janusgraph?pullRequest=1) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/janusgraph?pullRequest=1) it.</sub>

Commit:
Make optimisticLocking a StoreFeature. Silence noisy kcvslog test.

Signed-off-by: Alexander Patrikalakis <amcp@amazon.co.jp>

Commit:
Merge pull request #145 from amcp/testRefactor

Make optimisticLocking a StoreFeature. Silence noisy kcvslog test.

lindsayismith committed:
Merge pull request #116 from lindsayismith/bigtable-support-pr

Modifications to HBase storage to support Google Cloud Bigtable.  Includes documentation for connecting to Bigtable.

mbrukman committed:
Addressed PR review comments.

Signed-off-by: Misha Brukman <mbrukman@google.com>

mbrukman committed:
Merge pull request #115 from mbrukman/docs-cleanup

Docs cleanup

twilmes committed:
Updates to JanusGraphStepStrategy
* hasId folded into JanusGraphStep to prevent full scan
* Refactored JanusGraphStepStrategyTest to use TinkerPop strategy testing approach

mbrukman committed:
Fix capitalization of Bigtable.

Signed-off-by: Misha Brukman <mbrukman@google.com>

mbrukman committed:
Merge pull request #164 from mbrukman/docs-datamodel-bigtable

Fix capitalization of Bigtable in the docs

Commit:
Issue 162: Clean up Hadoop1 from doc and other places

mbrukman committed:
Merge pull request #165 from jerryjch/clean-up-hadoop1

Issue 162: Clean up Hadoop1 from doc and other places

twilmes committed:
Merge pull request #159 from twilmes/issue_155_hasId

Issue 155: Updated JanusGraphStepStrategy to fold hasId steps into JanusGraphStep

mbrukman committed:
Merge pull request #80 from rafatuita/configuration

Backward compatibility with Titan

Commit:
Fixed @return comments to remove javadoc warnings

Summary: Signed-off-by: Davide Bolcioni <dbolcion@uber.com>

Test Plan: mvn -pl 'janusgraph-core' package

Reviewers: kedar, alain, chris.lu, tunay

Reviewed By: tunay

Subscribers: jenkins

Differential Revision: https://code.uberinternal.com/D782525

mbrukman committed:
Merge pull request #168 from uber/return-warning

Fixed @return comments to remove javadoc warnings
